### PR TITLE
Fix iOS lifecycle

### PR DESCRIPTION
**Pull request by mapm14**
## Fix iOS Lifecycle

Currently, the app's iOS lifecycle wasn't working as expected, when the app was going to the background the Appyx Nodes were not updating their lifecycle states as expected. 

This PR solves the issue by making the iOS app to pass the app Lifecycle to the `iOSNodeHost`.

https://github.com/bumble-tech/appyx/assets/20324156/7bef8731-dee8-483d-a082-b9e792d728ea

## Check list

- [X] I have updated `CHANGELOG.md` if required.
- [X] I have updated documentation if required.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Please refer to [2.0.0-alpha10 – Migration guide](2.0.0-alpha10.md)
 - [#652](https://github.com/bumble-tech/appyx/pull/652) - KSP processor renamed from `mutable-ui-processor` to `appyx-processor`
 - [#654](https://github.com/bumble-tech/appyx/pull/654) - Renamings 
 - [#657](https://github.com/bumble-tech/appyx/pull/657) - Rename ParentNode & Node to Node and LeafNode 
+- [#660](https://github.com/bumble-tech/appyx/pull/660) - Make IosNodeHost to receive a Lifecycle class to fix iOS lifecycle problems
```
Comment (manuel-martos) [inline CHANGELOG.md:21]: Better to surround code related names with `` ` ``

File: demos/appyx-navigation/iosApp/iosApp/iOSApp.swift [hunk 2]
```diff
@@ -1,17 +1,52 @@
...
             .onOpenURL { incomingURL in
                 Main_iosKt.handleDeepLinks(url: incomingURL)
             }
+           .onChange(of: scenePhase) { newPhase in
+               switch newPhase {
+               case .background: appDelegate.lifecycleHolder.lifecycleHelper.created()
+               case .inactive: appDelegate.lifecycleHolder.lifecycleHelper.created()
+               case .active: appDelegate.lifecycleHolder.lifecycleHelper.resumed()
+               @unknown default: break
+               }
```
Comment (zsoltk) [inline demos/appyx-navigation/iosApp/iosApp/iOSApp.swift:29]: Could we move this from client code into some library code / do these in a way that copy-pasting it in every app is not needed?

Comment (mapm14) [reply demos/appyx-navigation/iosApp/iosApp/iOSApp.swift:29]: As discussed, we'll try to tackle this in a different PR :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ Please refer to [2.0.0-alpha10 – Migration guide](2.0.0-alpha10.md)
 - [#654](https://github.com/bumble-tech/appyx/pull/654) - Renamings 
 - [#657](https://github.com/bumble-tech/appyx/pull/657) - Rename ParentNode & Node to Node and LeafNode 
 - [#644](https://github.com/bumble-tech/appyx/pull/644) – Refactor AppyxComponent and application of draggable modifier 
+- [#660](https://github.com/bumble-tech/appyx/pull/660) - Make IosNodeHost to receive a Lifecycle class to fix iOS lifecycle problems
```
Comment (zsoltk) [inline CHANGELOG.md:21]: Let's move this to pending changes, alpha10 is already released

File: demos/appyx-navigation/iosApp/iosApp/iOSApp.swift [hunk 4]
```diff
@@ -1,17 +1,52 @@
...
         WindowGroup {
             ZStack {
                 Color.white.ignoresSafeArea(.all) // status bar color
-                ContentView()
+                ContentView(lifecycleHelper: appDelegate.lifecycleHolder.lifecycleHelper)
             }
             .onOpenURL { incomingURL in
                 Main_iosKt.handleDeepLinks(url: incomingURL)
             }
+           .onChange(of: scenePhase) { newPhase in
```
Comment (manuel-martos) [inline demos/appyx-navigation/iosApp/iosApp/iOSApp.swift:23]: Nitpick: fix indentation

File: demos/sandbox-appyx-navigation/iosApp/iosApp/iOSApp.swift [hunk 5]
```diff
@@ -1,13 +1,50 @@
...
+			.preferredColorScheme(.light)
+	        .onChange(of: scenePhase) { newPhase in
+                   switch newPhase {
+                   case .background: appDelegate.lifecycleHolder.lifecycleHelper.created()
+                   case .inactive: appDelegate.lifecycleHolder.lifecycleHelper.created()
+                   case .active: appDelegate.lifecycleHolder.lifecycleHelper.resumed()
+                   @unknown default: break
+                   }
+               }
 		}
```
Comment (manuel-martos) [inline demos/sandbox-appyx-navigation/iosApp/iosApp/iOSApp.swift:29]: Nitpick: fix indentation

**Comment by mapm14** [comment]
Fixed in https://github.com/bumble-tech/appyx/pull/670

**Pull request closed without merging**